Services/Component/classes/class.ilPluginAdmin.php

```diff
@@ -357,10 +357,54 @@ function hasLearningProgress($a_ctype, $a_cname, $a_slot_id, $a_pname)
 	* @return	bool
 	*/
 	function supportsExport($a_ctype, $a_cname, $a_slot_id, $a_pname)
-	{		
-		$this->getPluginData($a_ctype, $a_cname, $a_slot_id, $a_pname);		
+	{
+		$this->getPluginData($a_ctype, $a_cname, $a_slot_id, $a_pname);
 		return $this->data[$a_ctype][$a_cname][$a_slot_id][$a_pname]["supports_export"];
 	}
+
+	// cat-tms-patch start
+	/**
+	 * Get info for all plugins.
+	 *
+	 * @return 	array<string, array>
+	 */
+	public static function getAllPlugins() {
+		$cached_component = ilCachedComponentData::getInstance();
+		return $cached_component->getIlPluginById();
+	}
+
+	/**
+	 * Get info for all active plugins.
+	 *
+	 * @return 	array
+	 */
+	public static function getActivePlugins() {
+		$cached_component = ilCachedComponentData::getInstance();
+		$plugins = $cached_component->getIlPluginActive();
+		$buf = array();
+		foreach ($plugins as $slot => $plugs) {
+			$buf = array_merge($buf, $plugs);
+		}
+		return $buf;
+	}
+
+	/**
+	 * Check, if a plugin is active
+	 *
+	 * @param 	string 	$id 	id of the plugin
+	 * @return 	boolean
+	 */
+	public static function isPluginActive($id) {
+		assert('is_string($id)');
+		$cached_component = ilCachedComponentData::getInstance();
+		$plugs = $cached_component->getIlPluginById();
+		if(array_key_exists($id, $plugs) && $plugs[$id]['active']) {
+			return true;
+		}
+		return false;
+	}
+	// cat-tms-patch end
```

Comment:
I thought this already  was in the standard, but it isn't... Please include this in the PR.